utils/convert.py

```diff
@@ -1,52 +1,145 @@
 #!/usr/bin/python
 
 from __future__ import division
-from colormath.color_objects import LabColor, sRGBColor
+from colormath.color_objects import LabColor, sRGBColor, AppleRGBColor, BaseRGBColor, HSVColor
 from colormath.color_conversions import convert_color
-from pprint import pprint as pp
 import re
 import sys
 
-USAGE = """Convert from sRGB to Lab or from Lab to sRGB.
+USAGE = """Convert color(s) to all color spaces used by Selenized.
 
-The argument should either be a color (e.g. #c4d8df for sRGB, 23,-8,15 for Lab)
-or a path to a file with one color per line (empty lines allowed)."""
+The argument can be:
+  - a string representation of a color (e.g. c4d8df for sRGB, 23,-8,15 for Lab)
+  - a path to a file with one color per line (empty lines allowed)"""
 
-if len(sys.argv) != 2:
-    print USAGE
-    sys.exit()
 
-hex_num = r'([0-9a-fA-F]{2})'
-dec_num = r'(-?[0-9]{1,3})'
-hex_color_re = re.compile(r'#?' + hex_num*3)
-dec_color_re = re.compile(',\s*'.join([dec_num]*3))
+# Define some pretty printing for classes from colormath module.
+def lab_str(self):
+    return "{:>3.0f} {:>3.0f} {:>3.0f}".format(
+        self.lab_l, self.lab_a, self.lab_b
+    )
 
-def parse_color(color):
-    coords = hex_color_re.match(color)
+def hsv_str(self):
+    return "{:>3.0f} {:>3.0f} {:>3.0f}".format(
+        self.hsv_h, self.hsv_s*100, self.hsv_v*100
+    )
+
+# def rgb_str(self):
+#     return "{:>3.0f},{:>3.0f},{:>3.0f}".format(
+#         self.rgb_r*255, self.rgb_g*255, self.rgb_b*255
+#     )
+
+LabColor.__str__      = lab_str
+HSVColor.__str__      = hsv_str
+sRGBColor.__str__     = sRGBColor.get_rgb_hex
+AppleRGBColor.__str__ = AppleRGBColor.get_rgb_hex
+
+
+def parse_string(s):
+    """Extract 3 integers (hex or decimal) from a string.
+
+    Interprets strings beginning with a hash as hexadecimal,
+    strings separated with whitespace and/or commas as decimal."""
+
+    hex_num = r'([0-9a-fA-F]{2})'
+    dec_num = r'(-?[0-9]{1,3})'
+    hex_color_re = re.compile(r'#?' + hex_num*3)
+    dec_color_re = re.compile(',?\s*'.join([dec_num]*3))
+
+    coords = hex_color_re.match(s)
     if coords:
         base = 16
     else:
-        coords = dec_color_re.match(color)
+        coords = dec_color_re.match(s)
         base = 10
 
     parsed = [int(i, base) for i in coords.group(1, 2, 3)]
-    if base == 16:
-        return sRGBColor(*[i/255 for i in parsed])
-    else:
-        return LabColor(*parsed, illuminant='d50')
 
-def convert(color):
-    if isinstance(color, sRGBColor):
-        return convert_color(color, LabColor, target_illuminant='d50')
-    else:
-        return convert_color(color, sRGBColor)
+    return parsed, base
+
+
+class Color(object):
+    # some of the bright versions of colors are slightly out of RGB gamut -
+    # it's not a big deal, we just clamp them.
+    @staticmethod
+    def clamp(color):
```

Comment:
Why is it clamping only from one end? I mean, maybe there is no need to clamp negative values, but that is a bit unexpected for `clamp` function. Rename?

Yep, I'm a bit pesky, but I suppose that is what you expected from me :P